libpod/runtime_ctr.go

```diff
@@ -305,11 +306,19 @@ func (r *Runtime) setupContainer(ctx context.Context, ctr *Container) (_ *Contai
 				return nil, errors.Wrapf(err, "error retrieving named volume %s for new container", vol.Name)
 			}
 		}
-
+		var volOptions []VolumeCreateOption
 		logrus.Debugf("Creating new volume %s for container", vol.Name)
-
 		// The volume does not exist, so we need to create it.
-		volOptions := []VolumeCreateOption{WithVolumeName(vol.Name), WithVolumeUID(ctr.RootUID()), WithVolumeGID(ctr.RootGID())}
+		// Get appropriate user details.
+		if ctr.config.User == "" {
+			volOptions = []VolumeCreateOption{WithVolumeName(vol.Name), WithVolumeUID(ctr.RootUID()), WithVolumeGID(ctr.RootGID())}
+		} else {
+			uid, gid, _, err := chrootuser.GetUser(ctr.state.Mountpoint, ctr.config.User)
```

Comment:
Looking at this further, we aren't actually mounted when this call happens. We need to mount the container first, then immediately unmount after this finishes.